cirq-core/cirq/devices/fidelity_test.py

```diff
@@ -0,0 +1,210 @@
+import pytest
+import cirq
+from cirq.testing import assert_equivalent_op_tree
+from cirq.devices.fidelity import Fidelity, NoiseModelFromFidelity
+import numpy as np
+
+
+def test_invalid_arguments():
+    with pytest.raises(ValueError, match='At least one metric must be specified'):
+        Fidelity()
+
+    with pytest.raises(ValueError, match='xeb, pauli error, p00, and p11 must be between 0 and 1'):
+        Fidelity(p00=1.2)
+
+    with pytest.raises(ValueError, match='xeb, pauli error, p00, and p11 must be between 0 and 1'):
+        Fidelity(pauli_error=-0.2)
+
+    with pytest.raises(
+        ValueError,
+        match='Only one of xeb fidelity, pauli error, or decay constant should be defined',
+    ):
+        Fidelity(pauli_error=0.2, xeb_fidelity=0.5)
+
+    with pytest.raises(ValueError, match='A fidelity object must be specified'):
+        NoiseModelFromFidelity(None)
+
+
+def test_constructor():
+    xeb_fidelity = 0.95
+    p00 = 0.1
+    t1 = 200.0
```

Comment:
To be on the safe side, we should include a comment that this and other metrics in this file are purely for testing, and not necessarily representative of actual hardware behavior.